runtime/src/curated_grandpa.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A module for manually curated GRANDPA set.
+
+use {grandpa, system};
+use codec::Decode;
+use sr_primitives::traits::{As, Hash as HashT, BlakeTwo256, Zero};
+use rstd::prelude::*;
+
+
+pub trait Trait: grandpa::Trait {}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as CuratedGrandpa {
+		/// How often to shuffle the GRANDPA sets.
+		pub ShufflePeriod get(shuffle_period) build(|_| T::BlockNumber::sa(1024u64)): T::BlockNumber;
+	}
+}
+
+decl_module! {
+	/// curated GRANDPA set.
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		/// Changes the GRANDPA voter set.
+		fn set_voters(origin, voters: Vec<(T::SessionKey, u64)>) {
+			system::ensure_root(origin)?;
+			grandpa::Module::<T>::schedule_change(voters, T::BlockNumber::zero())?;
+		}
+
+		fn on_finalise(block_number: T::BlockNumber) {
+			// every so often shuffle the voters and issue a change.
+			let shuffle_period: u64 = Self::shuffle_period().as_();
+			if block_number.as_() % shuffle_period == 0 {
+				let mut seed = system::Module::<T>::random_seed().as_ref().to_vec();
+				seed.extend(b"grandpa_shuffling");
```

Comment:
Is this to ensure that the same seed value -> hash isn't used across multiple modules?

Reply:
i reckon so.

Reply:
yup.